conn.go

```diff
@@ -995,9 +995,11 @@ func (c *Conn) ReadPartitions(topics ...string) (partitions []Partition, err err
 			}
 
 			makeBrokers := func(ids ...int32) []Broker {
-				b := make([]Broker, len(ids))
-				for i, id := range ids {
-					b[i] = brokers[id]
+				var b []Broker
```

Comment:
This is pretty minor, but you can avoid potential allocations when growing the slice by allocating the space up-front:

```suggestion
				b := make([]Broker, 0, len(ids))
```